transcom/mymove
# MB-7423 Skeleton services counseling move details page

## Description

> ### User story
> 
> As a services counselor,
> I need to indicate counseling is finished,
> So that the TOO can access and approve the move.
> 
> In human language, the purpose of this ticket is to…make sure there’s a page that exists + make sure there’s a button where we can change the move state.
> 
> ### Acceptance Criteria
> 
> - There is a barebones SC Move Details page.
> 
> - There’s a Submit move details button at the top of the SC Move details page. For 
> 
> - After the user clicks that button:
> 
> - The button is no longer visible.
> 
> - System hits server endpoint to change move state (so it will be visible in TOO queue)
> 
> - Success message is displayed (or error, if the submit failed).

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?
- Reusing the `TXOMoveInfo` component. May need to create a new one so we can customize the header information.
- ~~Using the ReactUSWDS Alert component. In the design mockup, the alert component was smaller. What do you think?~~ Never mind, didn't realize the alert had a slim version :)
- Using route `/counseling/move/${MOVE_CODE}/details` for now until we implement the new role
- The endpoint to update the move status can only transition from `Needs Service Counseling` to `Service Counseling Completed`

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

Test
1. Log in as TOO on the office app
2. Navigate to `http://officelocal:3000/counseling/queue` services counseling queue page
3. Click on any move that has `Needs Services Counseling` status
4. Click on `Submit move details` button
5. Button will disappear
6. Success alert will appear up top

## Code Review Verification Steps


## References

* [Jira story](https://dp3.atlassian.net/browse/MB-7423) for this change
* [Design mockup](https://zpl.io/bWMpOXx)

## Screenshots

Move in `Needs service counseling` status
![image](https://user-images.githubusercontent.com/1522549/114128283-dfb85e00-98b0-11eb-825a-c2af755dab6e.png)

Submit move with slim alert box

https://user-images.githubusercontent.com/1522549/114219172-2e044600-991f-11eb-864f-73cbf94f4db6.mov
